### ISSUE TITLE
useful features

### ISSUE THREAD

**Issue by EGOKM** [microsoft/vscode-serial-monitor#175]
Type: Feature Request

Hello, I wanted to propose some hints for a "better" Serialmonitor as follows:

![shot-2024-04-09_13-56-18](https://github.com/microsoft/vscode-serial-monitor/assets/111112114/52f4cb89-a670-41cf-a488-f8dbc066799e)
![shot-2024-04-09_14-25-37](https://github.com/microsoft/vscode-serial-monitor/assets/111112114/4d3dc546-9309-4de4-85dd-a5e10f078416)

Probably it will be also useful for some other users ;-),

THX a lot, Enrico

**Comment by pkoevesdi** [comment]
I totaly also wish the hex output was supported by a column of printable characters. Otherwise I have no orientation in my output to identify my nonprintable bytes between the printable ones.

**Comment by EGOKM** [comment]
well, some kind of those ecape'd numers ( like \xHH or \n, \r, etc.) will be useful in text mode view, if there is some human readable stream over serial sometimes there are "stray" non-ascii bytes to be identified, or probably checksum's, for example: it may show "\x02Hello\x03", meaning there is an STX, then "Hello" follow'd by ETX ;-), same notation maybe also useful when sending some bytes in text mode ...

**Comment by pkoevesdi** [comment]
> well, some kind of those ecape'd numers ( like \xHH or \n, \r, etc.) will be useful in text mode view, if there is some human readable stream over serial sometimes there are "stray" non-ascii bytes to be identified, or probably checksum's, for example: it may show "\x02Hello\x03", meaning there is an STX, then "Hello" follow'd by ETX ;-), same notation maybe also useful when sending some bytes in text mode ...

But please not for "normal" letters inside words. I.e. please show "Menüpunkt", not "Men\xxpunkt". See also #168

**Comment by EGOKM** [comment]
ok, if You really want that there must be an additional setting for encoding ;-), if You want to see "ü"  here there are at least 2 choices: one is to use extended ASCII ( CodePage 437, "ü" is 0x81 here ) OR use UTF-encoding (0xc3 0xbc) ... more complicated but possible ;-)

**Comment by gcampbell-msft** [comment]
@EGOKM @pkoevesdi Could you elaborate more on the feedback for "a column of printable characters"? Currently the hex mode does print in columns, so I'm not sure what it is that you're requesting. Thanks!

**Comment by pkoevesdi** [comment]
Something like this: 
![th](https://github.com/microsoft/vscode-serial-monitor/assets/33196971/197b54bd-d941-461d-89bb-f1c97fc032a8)

**Comment by pkoevesdi** [comment]
> ok, if You really want that there must be an additional setting for encoding ;-)
Right. Or take the system default. Or recognize whether it is a 1-byte character set, assume it's ISO 8859-1, otherwise UTF-8.

**Comment by EGOKM** [comment]
@gcampbell-msft there is already an extension called Hex Editor ( original Microsoft ;-)  - same View ( but without Addresses ),
also, what about all the other thoughts ? Possible to make it ?

**Comment by gcampbell-msft** [comment]
@pkoevesdi Ah, I understand your request for the printable characters off to the side. This is definitely possible. We can put it on the backlog, however I'm not sure when we will get to it. Either way, we will mark this as "needs-more-upvotes" so that people can vote on the issue. 

@EGOKM I think this would be the main feature that we might consider, but again, we're not sure when we'd get to this feature. Thanks!

**Comment by EGOKM** [comment]
@gcampbell-msft well, regarding "we're not sure when we'd get to this feature" - it would be probably speed up things when You place the source code here, this way many people may contribute to it :-)
